src/main/java/ome/codecs/JPEGCodec.java

```diff
@@ -112,12 +122,13 @@ public byte[] decompress(RandomAccessInputStream in, CodecOptions options)
         in.seek(fp);
       }
 
-      b = ImageIO.read(new BufferedInputStream(new DataInputStream(in), 81920));
+      ImageInputStream stream = new MemoryCacheImageInputStream(new BufferedInputStream(in, 81920));
+      b = ImageIO.read(stream);
+      if (b == null)
+    	  throw new NullPointerException("ImageIO returned null when reading JPEG stream");
     }
     catch (IOException exc) {
-      // probably a lossless JPEG; delegate to LosslessJPEGCodec
-      in.seek(fp);
```

Comment:
The removal of this seek seems to be what causes tests to fail. The tests that failed in https://merge-ci.openmicroscopy.org/jenkins/job/BIOFORMATS-test-folder/213045/console pass locally if I reintroduce `in.seek(fp)` here.

Reply:
Added this back now